astroquery/wfau/core.py

```diff
@@ -0,0 +1,903 @@
+            request_payload['ra'] = C.ra.degree
+            request_payload['dec'] = C.dec.degree
+        elif sys == 'G':
+            C = commons.parse_coordinates(args[0]).transform_to(coord.Galactic)
+            request_payload['ra'] = C.l.degree
+            request_payload['dec'] = C.b.degree
+        return request_payload
+
+    def _verify_programme_id(self, pid, query_type='catalog'):
+        """
+        Verify the programme ID is valid for the query being executed.
+
+        Parameters
+        ----------
+        pid : int or str
+            The programme ID, either an integer (i.e., the # that will get passed
+            to the URL) or a string using the three-letter acronym for the
+            programme or its long name
+
+        Returns
+        -------
+        pid : int
+            Returns the integer version of the programme ID
+
+        Raises
+        ------
+        ValueError if the pid is 'all' and the query type is a catalog.
+        You can query all surveys for images, but not all catalogs.
+        """
+        if pid == 'all' and query_type == 'image':
+            return 'all'
+        elif pid == 'all' and query_type == 'catalog':
+            raise ValueError(
+                "Cannot query all catalogs at once. Valid catalogs are: {0}.\n"
+                "Change programmeID to one of these."
+                .format(",".join(self.programmes_short.keys())))
+        elif pid in self.programmes_long:
+            return self.programmes_long[pid]
+        elif pid in self.programmes_short:
+            return self.programmes_short[pid]
+        elif query_type != 'image':
+            raise ValueError("programme_id {0} not recognized".format(pid))
+
+    def _parse_system(self, system):
+        if system is None:
+            return 'J'
+        elif system.lower() in ('g', 'gal', 'galactic'):
+            return 'G'
+        elif system.lower() in ('j', 'j2000', 'celestical', 'radec'):
+            return 'J'
+
+    def get_images(self, coordinates, waveband='all', frame_type='stack',
+                   image_width=1 * u.arcmin, image_height=None, radius=None,
+                   database=None, programme_id=None,
+                   verbose=True, get_query_payload=False,
+                   show_progress=True):
+        """
+        Get an image around a target/ coordinates from a WFAU catalog.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string in which case it is resolved using online services or as
+            the appropriate `astropy.coordinates` object. ICRS coordinates
+            may also be entered as strings as specified in the
+            `astropy.coordinates` module.
+        waveband  : str
+            The color filter to download. Must be one of  ``'all'``, ``'J'``,
+            ``'H'``, ``'K'``, ``'H2'``, ``'Z'``, ``'Y'``, ``'Br'``].
```

Comment:
These are the ukidss filters. Either list both ukidss and vista, or the common ones, or don't provide a list at all.

Reply:
VSA has at least JHK; I think this is an inclusive list?